internal/app/machined/pkg/controllers/secrets/api.go

```diff
@@ -242,26 +241,23 @@ func (ctrl *APIController) reconcile(ctx context.Context, r controller.Runtime,
 		var endpointsStr []string
 
 		if !isControlplane {
-			endpointResource, err := r.Get(ctx, resource.NewMetadata(k8s.ControlPlaneNamespaceName, k8s.EndpointType, k8s.ControlPlaneEndpointsID, resource.VersionUndefined))
+			endpointResources, err := r.List(ctx, resource.NewMetadata(k8s.ControlPlaneNamespaceName, k8s.EndpointType, "", resource.VersionUndefined))
 			if err != nil {
-				if state.IsNotFoundError(err) {
-					continue
-				}
-
-				return fmt.Errorf("error getting endpoints resource: %w", err)
+				return fmt.Errorf("error getting endpoints resources: %w", err)
 			}
 
-			endpoints := endpointResource.(*k8s.Endpoint).TypedSpec()
+			var endpointAddrs k8s.EndpointList
```

Comment:
This change seems to necessitate public ip's for the Talos controlplane endpoints (apid + trustd services).   Without this change using a domain name for the canonical endpoint would permit resolution to a public IP of a load balancer that would then connect via private IP's to the endpoint services, including trustd on port 50001.  mydomain:50001 -> public ip port 50001 -> control plane registered private ip port 50001